homeassistant/components/yolink/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for yolink."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle yolink OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = ["create"]
+        return {"scope": " ".join(scopes)}
+
+    async def async_step_reauth(self, user_input=None) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
```

Comment:
We're missing test coverage for the config flow. It requires 100% test coverage. Please add tests for re-authentication and abort if already configured.

https://codecov.io/gh/home-assistant/core/compare/1be2438ef67c7f523654bdb849cbed5f4c865365...5e16f4ed18a89a388b24eeecd0fe5f017f3ddf96/diff

Reply:
I found that, i'm working on this